common/oatbox/cache/GcpTokenCacheItemPool.php

```diff
@@ -0,0 +1,253 @@
+
+    public const OPTION_PERSISTENCE = 'persistence';
+    public const OPTION_ENABLE_DEBUG = 'enableDebug';
+    public const OPTION_DISABLE_WRITE = 'disableWrite';
+    public const OPTION_TOKEN_CACHE_KEY = 'tokenCacheKey';
+
+    /** @var CacheItemPoolInterface */
+    private $cache;
+
+    /** @var bool */
+    private $isDebug;
+
+    /** @var bool */
+    private $isWritingDisabled;
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getItem($key)
+    {
+        $key = $this->getCacheKey();
+
+        $this->log(__METHOD__, [$key]);
+
+        $item = $this->getCache()->getItem($key);
+
+        $this->unSerializeIfNecessary($item);
+
+        return $item;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getItems(array $keys = [])
+    {
+        $keys = [$this->getCacheKey()];
+
+        $this->log(__METHOD__, $keys);
+
+        $items = $this->getCache()->getItems($keys);
+
+        foreach ($items as $item) {
+            $this->unSerializeIfNecessary($item);
+        }
+
+        return $items;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function hasItem($key)
+    {
+        $key = $this->getCacheKey();
+
+        $this->log(__METHOD__, [$key]);
+
+        return $this->getCache()->hasItem($key);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function clear()
+    {
+        $this->log(__METHOD__);
+
+        return $this->getCache()->clear();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function deleteItem($key)
+    {
+        $key = $this->getCacheKey();
+
+        $this->log(__METHOD__, [$key]);
+
+        return $this->getCache()->deleteItem($key);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function deleteItems(array $keys)
+    {
+        $keys = [$this->getCacheKey()];
+
+        $this->log(__METHOD__, $keys);
+
+        return $this->getCache()->deleteItems($keys);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function save(CacheItemInterface $item)
+    {
+        $this->log(__METHOD__);
+
+        if ($this->isWritingDisabled()) {
+            return true;
+        }
+
+        $this->serializeToPersist($item);
+
+        return $this->getCache()->save($item);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function saveDeferred(CacheItemInterface $item)
+    {
+        $this->log(__METHOD__);
+
+        if ($this->isWritingDisabled()) {
+            return true;
+        }
+
+        $this->serializeToPersist($item);
+
+        return $this->getCache()->saveDeferred($item);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function commit()
+    {
+        $this->log(__METHOD__);
+
+        if ($this->isWritingDisabled()) {
+            return true;
+        }
+
+        return $this->getCache()->commit();
+    }
+
+    private function log(string $method, array $keys = []): void
+    {
+        if ($this->isDebug()) {
+            $this->getLogger()->debug(
```

Comment:
What is the purpose? If logger has been configured to keep `debug` level - we are already fine; 
Most of the time production envs will not log anything, even if this particular service is in debug mode, as logger verbosity used to at about ~ `warn` level

Reply:
This is a good point, indeed we do not need this if. I will remove! Thanks for the heads up!